src/client/dfuse/pil4dfs/hook.c

```diff
@@ -114,16 +150,19 @@ read_map_file(char **buf, const char str_target[], char **pos)
 	/* Keep reading the file until finish the whole file. Increase the buffer size if needed. */
 	while (complete == 0) {
 		buffer_full = false;
-		*buf = malloc(max_read_size + 1);
+		D_ALLOC(*buf, max_read_size + 1);
 		if (*buf == NULL) {
-			printf("Failed allocate memory (%d bytes).\nQuit\n", max_read_size + 1);
+			D_FATAL("Failed allocate memory (%d bytes). %d (%s)\n", max_read_size + 1,
```

Comment:
Overall I think it would be much better, and essential for DAOS-14209 and testing, to have a common error message that is logged by `pil4dfs` if it can't start for any reason.  To that end I'd probably have these errors all be DS_ERROR(), and then a `goto fatal` where D_FATAL logs a common string that we can then put in the documentation and test code to detect for startup failure.

Not a regression however so I won't block this work on it.